pkg/server/jsonrpc.go

```diff
@@ -23,18 +23,24 @@ type JSONRPC interface {
 	communicate(buf []byte) (io.Reader, error)
 }
 
+// TODO: rename to remove unix since now this is generic (last commit)
 type unixSocketJSONRPC struct {
 	transport string
 	socket    *string
 	id        uint64
 }
 
 // NewUnixSocketJSONRPC creates a new instance of JSONRPC which is capable to
-// interact with unix domain socket
+// interact with either unix domain socket, e.g.: /var/tmp/spdk.sock
+// or with tcp connection ip and port tuple, e.g.: 10.1.1.2:1234
 func NewUnixSocketJSONRPC(socketPath string) JSONRPC {
-	// TODO: get transport from socketPath automatically
+	protocol := "tcp"
```

Comment:
I'd check if `socketPath` is empty or not and raise panic(return error) if it is

Reply:
Agree #254